File: app/src/test/java/com/mgaetan89/showsrage/model/Episode_GetStatusForMenuIdTest.java
package com.mgaetan89.showsrage.model;

import com.mgaetan89.showsrage.R;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class Episode_GetStatusForMenuIdTest {
	@Parameterized.Parameter(1)
	public String episodeStatus;

	@Parameterized.Parameter(0)
	public int menuId;

	@Test
	public void getStatusForMenuId() {
		assertThat(Episode.getStatusForMenuId(this.menuId)).isEqualTo(this.episodeStatus);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{R.id.menu_episode_set_status_failed, "failed"},
				{R.id.menu_episode_set_status_ignored, "ignored"},
				{R.id.menu_episode_set_status_skipped, "skipped"},
				{R.id.menu_episode_set_status_wanted, "wanted"},
				{R.id.menu_change_quality, null},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/model/Episode_GetStatusForMenuIdTest.kt
package com.mgaetan89.showsrage.model

import com.mgaetan89.showsrage.R
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class Episode_GetStatusForMenuIdTest {
    @Parameterized.Parameter(1)
    var episodeStatus: String? = null

    @Parameterized.Parameter(0)
    var menuId: Int = 0

    @Test
    fun getStatusForMenuId() {
        assertThat(Episode.getStatusForMenuId(this.menuId)).isEqualTo(this.episodeStatus)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any?>> {
            return listOf(
                    arrayOf<Any?>(R.id.menu_episode_set_status_failed, "failed"),
                    arrayOf<Any?>(R.id.menu_episode_set_status_ignored, "ignored"),
                    arrayOf<Any?>(R.id.menu_episode_set_status_skipped, "skipped"),
                    arrayOf<Any?>(R.id.menu_episode_set_status_wanted, "wanted"),
                    arrayOf<Any?>(R.id.menu_change_quality, null)
            )
        }
    }
}
